src/services/identity/LaunchesService.ts

```diff
@@ -159,6 +161,41 @@ export class LaunchesService {
       throw siteIdResult.error
     }
 
+    try {
+      // Check if association already exists
+      const getDomainAssociationCommandInput = this.launchClient.createGetDomainAssociationCommandInput(
+        appIdResult.value,
+        domainName
+      )
+
+      const getDomainAssociationResult = await this.launchClient.sendGetDomainAssociationCommand(
+        getDomainAssociationCommandInput
+      )
+      const hasDomainAssociationFailed =
+        getDomainAssociationResult?.domainAssociation?.domainStatus === "FAILED"
```

Comment:
Refer [here](https://repost.aws/questions/QUxsRQ8_2KSJ68j-Dk5M7Xvg/amplify-domain-creation-statuses-what-do-they-mean) for types of domain association result